advanced/src/index.js

```diff
@@ -0,0 +1,180 @@
+import React, {Component} from 'react'
+import ReactDOM from 'react-dom'
+import {
+  NavLink,
+  Link,
+  BrowserRouter as Router,
+  Route,
+  Switch,
+  Redirect
+} from 'react-router-dom'
+import { ApolloProvider } from 'react-apollo'
+import { HttpLink, InMemoryCache, ApolloClient } from 'apollo-client-preset'
+import { setContext } from 'apollo-link-context'
+import { WebSocketLink } from 'apollo-link-ws'
+import { ApolloLink, split } from 'apollo-link'
+import { getMainDefinition } from 'apollo-utilities'
+
+
+import FeedPage from './components/FeedPage'
+import DraftsPage from './components/DraftsPage'
+import CreatePage from './components/CreatePage'
+import DetailPage from './components/DetailPage'
+import LoginPage from './components/LoginPage'
+import SignupPage from './components/SignupPage'
+import PageNotFound from './components/PageNotFound'
+import LogoutPage from './components/LogoutPage'
+import { AUTH_TOKEN } from './constant'
+
+import 'tachyons'
+import './index.css'
+
+const httpLink = new HttpLink({ uri: 'http://localhost:4000/' })
+
+const middlewareLink = new ApolloLink((operation, forward) => {
+  // get the authentication token from local storage if it exists
+  const tokenValue = localStorage.getItem(AUTH_TOKEN)
+  // return the headers to the context so httpLink can read them
+  operation.setContext({
+    headers: {
+      Authorization: tokenValue ? `Bearer ${tokenValue}` : "",
+    }
+  })
+  return forward(operation)
+})
+
+// Authenticated httplink
+const httpLinkAuth = middlewareLink.concat(httpLink)
+
+const wsLink = new WebSocketLink({
+  uri: `wss://localhost:4000/`,
+  options: {
+    reconnect: true,
+    connectionParams: {
+      Authorization: `Bearer ${localStorage.getItem(AUTH_TOKEN)}`
+    }
+  }
+})
+
+const link = split(
+  // split based on operation type
+  ({ query }) => {
+    const { kind, operation } = getMainDefinition(query)
+    return kind === 'OperationDefinition' && operation === 'subscription'
+  },
+  wsLink,
+  httpLinkAuth,
+)
+
+// apollo client setup
+const client = new ApolloClient({
+  link: ApolloLink.from([link]),
+  cache: new InMemoryCache(),
+  connectToDevTools: true
+})
+
+  const token = localStorage.getItem (AUTH_TOKEN)
+
+  const ProtectedRoute = ({component: Component, token, ...rest}) => {
```

Comment:
Could you possibly clean this up a bit? It looks a little hard on the eyes. Something like:

```js
const ProtectedRoute = ({component: Component, token, ...rest}) => {
  return token
    ? (<Route {...rest} render={matchProps => (<Component {...matchProps} />)} />)
    : (<Redirect to="/login" />)
};
```

Reply:
I'm noticing a lot of these changes could be done automatically if you are using Prettier. If you need some help setting it up, reach out to me on Slack @manjula91. It could speed up your development time some too.